codegen-server/python/src/main/kotlin/software/amazon/smithy/rust/codegen/server/python/smithy/PythonServerSymbolProvider.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.server.python.smithy
+
+import software.amazon.smithy.codegen.core.Symbol
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.Shape
+import software.amazon.smithy.rust.codegen.rustlang.RustMetadata
+import software.amazon.smithy.rust.codegen.rustlang.RustType
+import software.amazon.smithy.rust.codegen.smithy.RuntimeType
+import software.amazon.smithy.rust.codegen.smithy.RustSymbolProvider
+import software.amazon.smithy.rust.codegen.smithy.WrappingSymbolProvider
+import software.amazon.smithy.rust.codegen.smithy.meta
+import software.amazon.smithy.rust.codegen.smithy.rustType
+
+/**
+ * Input / output / error structures can refer to complex types like the ones implemented inside
+ * `aws_smithy_types` (a good example is `aws_smithy_types::Blob`).
+ * `aws_smithy_http_server_python::types` wraps those types that do not implement directly the
+ * `pyo3::PyClass` trait and cannot be share safely to Python, providing an idiomatic Python / Rust API.
+ *
+ * This symbol provider ensures types not implemeting `pyo3::PyClass` are swapped with their wrappers from
+ * `aws_smithy_http_server_python::types`.
+ */
+class PythonServerSymbolProvider(private val base: RustSymbolProvider, private val model: Model) :
+    WrappingSymbolProvider(base) {
+
+    private val runtimeConfig = config().runtimeConfig
+
+    /**
+     * Convert a shape to a Symbol.
+     *
+     * Swap the symbol if the shape's symbol does not implement `pyo3::PyClass`.
+     */
+    override fun toSymbol(shape: Shape): Symbol {
+        return when (base.toSymbol(shape).rustType()) {
+            RuntimeType.Blob(runtimeConfig).toSymbol().rustType() -> {
```

Comment:
I don't understand..

Reply:
The issue here was that since this symbol provider was executed before the `MetadataSymbolProvider`, causing runtime symbols defined here to come without the necessary metadata. Moving this symbol provider down the stack solved this issue and allowed me to just use `PythonServerRuntimeType.Blob.toSymbol()`.